### ISSUE TITLE
resuming doesnt work!

### ISSUE THREAD

**Issue by Coderx7** [D-X-Y/ResNeXt-DenseNet#19]
Hi, After several days of training on imageNet I faced a power outage and had to resume training. 
I simply used the `--resume` switch with the last checkpoint and rerun the training. However, I noticed the loss is very high and it seems it does not load the parameters correctly. I have no idea what is happening here! 
Here is the sample of the log just before power outage happens : 
```
==>>[2018-06-24 10:18:56] [Epoch=083/150] [Need: 76:10:35] [learning_rate=0.001000] [Best : Accuracy(T1/T5)=65.93/86.76, Error=34.07/13.24]
Epoch: [83][0/10010]	Time 4.381 (4.381)	Data 4.204 (4.204)	Loss 1.8135 (1.8135)	Prec@1 58.594 (58.594)	Prec@5 82.031 (82.031)
Epoch: [83][200/10010]	Time 0.401 (0.418)	Data 0.001 (0.021)	Loss 1.9176 (1.6593)	Prec@1 55.469 (61.979)	Prec@5 77.344 (82.467)
Epoch: [83][400/10010]	Time 0.398 (0.409)	Data 0.000 (0.011)	Loss 1.5161 (1.6541)	Prec@1 63.281 (61.886)	Prec@5 85.938 (82.573)
Epoch: [83][600/10010]	Time 0.400 (0.406)	Data 0.000 (0.007)	Loss 1.4733 (1.6492)	Prec@1 64.062 (62.100)	Prec@5 85.938 (82.597)
Epoch: [83][800/10010]	Time 0.414 (0.405)	Data 0.000 (0.006)	Loss 1.6637 (1.6474)	Prec@1 62.500 (62.121)	Prec@5 78.906 (82.592)
Epoch: [83][1000/10010]	Time 0.401 (0.404)	Data 0.000 (0.005)	Loss 1.6384 (1.6505)	Prec@1 63.281 (62.057)	Prec@5 85.938 (82.540)
Epoch: [83][1200/10010]	Time 0.405 (0.403)	Data 0.000 (0.004)	Loss 1.5951 (1.6491)	Prec@1 61.719 (62.043)	Prec@5 82.031 (82.559)
Epoch: [83][1400/10010]	Time 0.399 (0.403)	Data 0.001 (0.003)	Loss 1.5070 (1.6480)	Prec@1 64.844 (62.062)	Prec@5 81.250 (82.552)
Epoch: [83][1600/10010]	Time 0.395 (0.402)	Data 0.000 (0.003)	Loss 1.2925 (1.6455)	Prec@1 71.094 (62.141)	Prec@5 88.281 (82.604)
Epoch: [83][1800/10010]	Time 0.401 (0.402)	Data 0.000 (0.003)	Loss 1.7567 (1.6462)	Prec@1 55.469 (62.089)	Prec@5 79.688 (82.583)
Epoch: [83][2000/10010]	Time 0.394 (0.402)	Data 0.000 (0.003)	Loss 1.4647 (1.6442)	Prec@1 69.531 (62.143)	Prec@5 86.719 (82.574)
Epoch: [83][2200/10010]	Time 0.398 (0.401)	Data 0.000 (0.002)	Loss 2.0746 (1.6443)	Prec@1 54.688 (62.147)	Prec@5 75.000 (82.573)
Epoch: [83][2400/10010]	Time 0.400 (0.401)	Data 0.000 (0.002)	Loss 1.6010 (1.6446)	Prec@1 63.281 (62.125)	Prec@5 79.688 (82.553)
Epoch: [83][2600/10010]	Time 0.408 (0.401)	Data 0.000 (0.002)	Loss 1.6674 (1.6434)	Prec@1 57.031 (62.152)	Prec@5 82.812 (82.586)
Epoch: [83][2800/10010]	Time 0.405 (0.401)	Data 0.000 (0.002)	Loss 1.2825 (1.6438)	Prec@1 67.969 (62.133)	Prec@5 88.281 (82.579)
Epoch: [83][3000/10010]	Time 0.401 (0.401)	Data 0.000 (0.002)	Loss 1.5878 (1.6436)	Prec@1 63.281 (62.156)	Prec@5 80.469 (82.588)
Epoch: [83][3200/10010]	Time 0.393 (0.401)	Data 0.000 (0.002)	Loss 1.5673 (1.6442)	Prec@1 66.406 (62.131)	Prec@5 81.250 (82.593)
Epoch: [83][3400/10010]	Time 0.398 (0.401)	Data 0.000 (0.002)	Loss 1.6198 (1.6443)	Prec@1 61.719 (62.119)	Prec@5 80.469 (82.599)
Epoch: [83][3600/10010]	Time 0.396 (0.401)	Data 0.000 (0.002)	Loss 1.4174 (1.6435)	Prec@1 60.938 (62.127)	Prec@5 85.938 (82.597)
Epoch: [83][3800/10010]	Time 0.403 (0.401)	Data 0.000 (0.002)	Loss 1.9573 (1.6424)	Prec@1 64.844 (62.161)	Prec@5 78.906 (82.604)
Epoch: [83][4000/10010]	Time 0.396 (0.400)	Data 0.001 (0.002)	Loss 1.7674 (1.6429)	Prec@1 64.062 (62.157)	Prec@5 80.469 (82.606)
Epoch: [83][4200/10010]	Time 0.403 (0.400)	Data 0.000 (0.001)	Loss 1.7971 (1.6425)	Prec@1 60.938 (62.152)	Prec@5 82.812 (82.607)
Epoch: [83][4400/10010]	Time 0.400 (0.400)	Data 0.000 (0.001)	Loss 1.7141 (1.6433)	Prec@1 61.719 (62.135)	Prec@5 79.688 (82.604)
Epoch: [83][4600/10010]	Time 0.401 (0.400)	Data 0.000 (0.001)	Loss 1.5213 (1.6427)	Prec@1 67.188 (62.159)	Prec@5 88.281 (82.616)
Epoch: [83][4800/10010]	Time 0.403 (0.400)	Data 0.000 (0.001)	Loss 1.9228 (1.6429)	Prec@1 55.469 (62.154)	Prec@5 76.562 (82.614)
Epoch: [83][5000/10010]	Time 0.404 (0.400)	Data 0.000 (0.001)	Loss 1.5705 (1.6437)	Prec@1 60.938 (62.142)	Prec@5 84.375 (82.609)
Epoch: [83][5200/10010]	Time 0.411 (0.400)	Data 0.000 (0.001)	Loss 1.5285 (1.6428)	Prec@1 63.281 (62.170)	Prec@5 85.156 (82.622)
Epoch: [83][5400/10010]	Time 0.397 (0.400)	Data 0.000 (0.001)	Loss 1.7705 (1.6431)	Prec@1 60.156 (62.165)	Prec@5 82.031 (82.614)
Epoch: [83][5600/10010]	Time 0.394 (0.400)	Data 0.000 (0.001)	Loss 1.4945 (1.6432)	Prec@1 60.938 (62.156)	Prec@5 82.031 (82.616)
Epoch: [83][5800/10010]	Time 0.398 (0.400)	Data 0.000 (0.001)	Loss 1.7673 (1.6427)	Prec@1 59.375 (62.160)	Prec@5 80.469 (82.626)
```

and this is after I resumed : 
```
==>>[2018-06-24 11:22:55] [Epoch=083/150] [Need: 00:00:00] [learning_rate=0.100000] [Best : Accuracy(T1/T5)=65.93/0.00, Error=34.07/100.00]
Epoch: [83][0/10010]	Time 7.624 (7.624)	Data 3.742 (3.742)	Loss 1.9200 (1.9200)	Prec@1 57.031 (57.031)	Prec@5 76.562 (76.562)
Epoch: [83][200/10010]	Time 0.414 (0.456)	Data 0.000 (0.019)	Loss 5.5993 (6.0936)	Prec@1 2.344 (2.651)	Prec@5 10.156 (8.302)
Epoch: [83][400/10010]	Time 0.400 (0.435)	Data 0.000 (0.010)	Loss 5.1322 (5.7781)	Prec@1 9.375 (3.943)	Prec@5 22.656 (11.898)
Epoch: [83][600/10010]	Time 0.396 (0.429)	Data 0.000 (0.007)	Loss 4.7684 (5.5425)	Prec@1 10.938 (5.486)	Prec@5 33.594 (15.561)
Epoch: [83][800/10010]	Time 0.404 (0.426)	Data 0.000 (0.005)	Loss 4.7264 (5.3447)	Prec@1 8.594 (7.054)	Prec@5 31.250 (18.790)

```
As you can see the loss originally is 1 but when I resume it becomes 5! its as if it doesn't load anything and starts from the beginning! 
I tried this 3 times and I get this result every time. I also used the `start_epoch`(set it to 83 and 82), and that doesn't help either. 
What should I do?

**Comment by D-X-Y** [comment]
Hi, from the log, it seems that the learning rate is not correct. After resume, the learning rate should be 0.001 rather than 0.1

**Comment by Coderx7** [comment]
Thanks alot. Shouldn't the learning rate (basically all the optimization statistics) be retained and get resumed automatically? what should I do then to address this properly>? i mean where in the code should I have the changes?

**Comment by D-X-Y** [comment]
Hi, are you using "imagenet_train.py" to train on ImageNet? My code seems it will not print "[Best : Accuracy(T1/T5)=65.93/0.00, Error=34.07/100.00]"? Did you modify some codes?

**Comment by Coderx7** [comment]
yeah, that's the same code, however, I added some more prints from your `main.py` (CIFAR10/100 training script) to make the stats look better. here is the whole deal with the changes I made: [imagenet_train.py](https://gist.github.com/Coderx7/3090242a013464c5af8931a14250319e)

**Comment by D-X-Y** [comment]
I see, you use the scheduler from PyTorch 0.4.0. You should also save the state of scheduler in the checkpoint, and resume it from checkpoint. My code is based on PyTorch 0.3.0, which does not include the scheduler component. Therefore, you got the wrong learning rate.

**Comment by Coderx7** [comment]
@D-X-Y : Wow man! Thanks a gazillion times! God bless you, this has been haunting me for a month now and no one in pytorch discussion, nor Reddit could answer me!

**Comment by D-X-Y** [comment]
haha, you are welcome~
